src/java/com/puppetlabs/http/client/impl/ClientMetricFilter.java

```diff
@@ -0,0 +1,64 @@
+package com.puppetlabs.http.client.impl;
+
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.Metric;
+import org.apache.commons.lang3.StringUtils;
+
+public class ClientMetricFilter {
```

Comment:
One other option might be to just create a single `MetricsFilter` class, `MetricFilterMatchingName` or something like that.  The constructor could take in a `String` object and use that object to do the `equals` comparison in a `matches` method callback.  Could then move the logic to construct the appropriate sub-namespace arrays back into calling `getClientMetrics*` methods in `Metrics` class?